### DISCUSSION TITLE
Using Foundry's makeAddr() for Contract Deployment: Is it possible to deploy main contract using dummy addresses?

### DISCUSSION THREAD

**Discussion by Shourya4641** [Cyfrin/foundry-full-course-cu#2853] [Q&A] [answered]
# Using Foundry's makeAddr() for Contract Deployment: Is it possible to deploy main contract using dummy addresses?

## Question

I'm working with Foundry for testing smart contracts and I'm curious about extending the usage of `makeAddr()` beyond just testing scenarios.

Currently, I know we can create dummy addresses for testing using:
```solidity
address alice = makeAddr("alice");
address bob = makeAddr("bob");
```

What I want to know is:
1. Can `makeAddr()` be used to create a dummy address for deploying our main contract (not just for testing)?
2. If yes, what would be the best approach to implement this?
3. Are there any security implications or limitations we should know?

I'm specifically interested in this because it would help in creating consistent deployment addresses across different environments or it would make our deployment process more deterministic.

Has anyone tried this before or knows if it is feasible? I would greatly appreciate any insights or alternative approaches.

### Environment
- Foundry Version: 0.2.0
- Solidity Version: 0.8.24

#foundry #smart-contracts #ethereum #testing

**Comment by EngrPips** [comment, marked as answer]
Hello @Shourya4641, Honestly I haven't tried this before and I don't know it is a thing but I assume the `vm.makeAddr()` is created by the Foundry mainly for the purpose of creating random address for testing and not for deploying contracts, If you are looking for a Foundry cheatcode which will help you deploy a contract to specific address you can look into Foundry `vm.etch` cheatcode https://book.getfoundry.sh/cheatcodes/etch
